### DISCUSSION TITLE
IN style operator

### DISCUSSION THREAD

**Discussion by sjblack** [alirezanet/Gridify#133] [Q&A] [answered]
Hello,

I'm wondering if there is anyway of producing the following linq query.

var states = new List<Guid> {...};

x => states.Contains(x.State);

**Comment by alirezanet** [comment]
Hi @sjblack,

Gridify doesn't have a built-in `IN` operator yet _(`v2.12.0`)_, but you can create your own custom operator if you need:
e.g

``` csharp
GridifyGlobalConfiguration.CustomOperators.Register(new InOperator());

var lst = new List<TestClass>() {
	new(new List<int> { 1,3,5,7,9 }), 
	new(new List<int> { 2,4,6,8,10 }) 
}.AsQueryable();

var x1 = lst.ApplyFiltering("Ids#=5");
var x2 = lst.Where(q => q.Ids.Contains(5) );

// x1 and x2 are exactly the same using custom `InOperator`

// ------------ 
record TestClass (IList<int> Ids);

class InOperator : IGridifyOperator
{
   public string GetOperator() => "#=";
   public Expression<OperatorParameter> OperatorHandler()
   {
       return (prop, value) => ((IList)prop).Contains(Convert.ToInt32(value));
   }
}

```
[More info](https://alirezanet.github.io/Gridify/guide/filtering.html#custom-operators)

**Comment by sjblack** [comment]
Thank you for the quick reply.

I did give this a go, its a slightly different scenario to the example you have provided. It's more like this in sql:

```sql
'SELECT State FROM Tasks WHERE State IN (guid1, guid2)'
```

and Linq:

```C#
class Task
{
   Guid State { get; set; }
}

listOfGuids = List<Guid> { guid1, guid2 }

Tasks.Where(x => listOfGuids.Contains(x.State));
```
If I create the custom operator as follows its doesn't seem to play ball:

````C#
class InOperator : IGridifyOperator
{
   public string GetOperator() => "#=";
   public Expression<OperatorParameter> OperatorHandler()
   {
       return (prop, value) => ((IList)value).Contains(Guid.Parse(prop));
   }
}
````
Worth mentioning I used a converter on the mapper  to create the list of guids from a string representation.

Appreciate the help!

**Reply by alirezanet** [reply]
Are you using Entity framework?

**Reply by sjblack** [reply]
Yes I am, at the moment EF6.

**Reply by alirezanet** [reply]
At this moment I don't think there is any easy way to pass list-style values to gridify, but thank you for bringing this up. I'll try to find a way or maybe add another operator.

**Reply by alirezanet** [reply, marked as answer]
I've also created an issue for this #135 if you want to follow.

**Reply by sjblack** [reply]
Cool, I'll keep my eye on it. Thanks very much!

**Comment by alirezanet** [comment]
in v2.13.1, we can use custom operators for this purpose, more info https://github.com/alirezanet/Gridify/issues/135

https://alirezanet.github.io/Gridify/guide/filtering.html#custom-operators
